tool/tsh/tsh_test.go

```diff
@@ -37,46 +40,166 @@ import (
 	"github.com/gravitational/teleport/lib/tlsca"
 	"github.com/gravitational/teleport/lib/utils"
 	"github.com/gravitational/teleport/tool/tsh/common"
-	"github.com/stretchr/testify/require"
 
-	"gopkg.in/check.v1"
+	"github.com/stretchr/testify/require"
 )
 
-// bootstrap check
-func TestTshMain(t *testing.T) {
-	check.TestingT(t)
-}
+const staticToken = "test-static-token"
 
-// register test suite
-type MainTestSuite struct{}
+var randomLocalAddr = utils.NetAddr{AddrNetwork: "tcp", Addr: "127.0.0.1:0"}
 
-var _ = check.Suite(&MainTestSuite{})
-
-func (s *MainTestSuite) SetUpSuite(c *check.C) {
+func TestOIDCLogin(t *testing.T) {
 	utils.InitLoggerForTests(testing.Verbose())
 	os.RemoveAll(client.FullProfilePath(""))
+
+	defer os.RemoveAll(client.FullProfilePath(""))
```

Comment:
I found the the removal of the default tsh login extremely hideous when working with the external environments (after a successful `tsh login`) and with (unit) tests - I think we should make an effort with replacing this with a temporary location as is otherwise customary for testing. WDYT?

Reply:
I agree, but currently `tsh` behavior does not allow custom profile paths.  It should, but, since the clearing behavior predates this PR, I'd prefer to add that functionality as a separate PR with a decent amount of manual testing to ensure no regressions.